requester.py

```diff
@@ -0,0 +1,99 @@
+import requests
+
+# For now request works only for default '24h' forecast period
+
+# weatherapi.com API
+
+
+def request_weatherapi(keys_dict, args):
+    url_wheatherapi = 'https://api.weatherapi.com/v1/forecast.json'
+    API_KEY_wheatherapi = keys_dict['wheatherapi.com']
+
+    params_wheatherapi = dict(key=API_KEY_wheatherapi, q=args.location, days=1)
+    resonse_wheatherapi = requests.get(url_wheatherapi, params=params_wheatherapi)
+
+    # check if lack of proper request response is due to 403 error
+    if resonse_wheatherapi.status_code == 403:
+        print("Supplied API key for wheatherapi.com is wrong.")
+        exit()
+
+    resonse_wheatherapi_dict = resonse_wheatherapi.json()
+    resonse_wheatherapi_hours = resonse_wheatherapi_dict['forecast']['forecastday'][0]['hour']
+
+    # extracting needed information from request response
+    hours_wheatherapi = []
+    temp_c_wheatherapi = []
+    rain_wheatherapi = []
+    for item in resonse_wheatherapi_hours:
+        hours_wheatherapi.append(item["time"])
+        temp_c_wheatherapi.append(item["temp_c"])
+        rain_wheatherapi.append(item['precip_mm'])
+
+    # get latitute and longitute of the location
+    latitude = resonse_wheatherapi_dict['location']['lat']
+    longitude = resonse_wheatherapi_dict['location']['lon']
+
+    weatherapi_dict = {'hours': hours_wheatherapi, 'temp': temp_c_wheatherapi,
+                       'rain': rain_wheatherapi, 'latitude': latitude, 'longitude': longitude}
+
+    return weatherapi_dict
+
+# open-meteo.com API
+
+
+def request_openmeteo(latitude, longitude):
+    url_openmeteo = "https://api.open-meteo.com/v1/forecast"
+    params_openmeteo = {
+        "latitude": latitude,
+        "longitude": longitude,
+        "hourly": ["temperature_2m", "precipitation"],
+        "forecast_days": 1
+    }
+
+    response_openmeteo = requests.get(url_openmeteo, params=params_openmeteo)
+    response_openmeteo_dict = response_openmeteo.json()
+
+    hours_openmeteo = response_openmeteo_dict['hourly']['time']
+    temp_c_openmeteo = response_openmeteo_dict['hourly']['temperature_2m']
+    rain_openmeteo = response_openmeteo_dict['hourly']['precipitation']
+
+    openmeteo_dict = {'hours': hours_openmeteo, 'temp': temp_c_openmeteo,
+                      'rain': rain_openmeteo, 'latitude': latitude, 'longitude': longitude}
+
+    return openmeteo_dict
+
+# meteostat.net
+
+
+def request_meteostat(keys_dict, latitude, longitude, current_date):
+    url_meteostat = 'https://meteostat.p.rapidapi.com/point/hourly'
+    API_KEY_meteostat = keys_dict['meteostat.net']
+    # current_date = hours_wheatherapi[1].split(' ')[0]
+
+    params_meteostat = {
+        "lat": latitude,
+        "lon": longitude,
+        "start": current_date,  # current day obtained from first api request
```

Comment:
Why the `current_date` does not work in this case? In general, it makes sense to pass it over to all those functions because the user chooses it. Even if it's hardcoded to 24h as for now, please hardcode it in the Main component and pass it to those. This way you'll be ready for the future.

Reply:
I created current_date variable, but I am using it only for the last API request. I checked other API services but I did not find a way to add it to the other's requests easily. I will deal with it later.

Reply:
Again, often things like that left for later are not done or easily forgotten. Here it's a matter of passing one arg to all the functions, it should not be complicated. Let's talk about it at the upcoming meeting.

Reply:
@deerhunter4 ok it will not be used in the default `24h` period case, but what if the user chooses to use another period time? For E.g. in the `request_openmeteo` method, I see you use the `forecast_days` param in the request, so this could and should be doable. And even if you don't want to use that param now (which is in contrary to the requirements we have, and that requires yet another ticket for handling that issue later), please at least pass it for now with a comment saying this part will be dealt with later (and let me know explicitly so that I will create a task for it). Again:

- let's talk about it at the upcoming meeting if you're not sure
- before the meeting please at least try to handle that, and if not successful, pass at least a placeholder param for now with a comment
- we can also try to figure that out together if you want and by that occasion practice some [pair programming](https://en.wikipedia.org/wiki/Pair_programming)
- all that because, as I already said "things like that left for later are not done or easily forgotten";)

Reply:
@KrzysiekJodlowski I think that pair programming was a success. Everything looks ready for a merge.